velox/exec/HashTable.cpp

```diff
@@ -221,12 +240,14 @@ class ProbeState {
     __builtin_prefetch(group_ + firstKey);
   }
 
-  void eraseHit(uint8_t* tags) {
-    const auto kEmptyGroup = BaseHashTable::TagVector::broadcast(0);
-    auto empty = simd::toBitMask(tagsInTable_ == kEmptyGroup);
+  void eraseHit(uint8_t* tags, int64_t& numTombstones) {
+    const auto kEmptyGroup = BaseHashTable::TagVector::broadcast(kEmptyTag);
+    const bool hasEmptyGroup =
+        simd::toBitMask(tagsInTable_ == kEmptyGroup) != 0;
 
     BaseHashTable::storeTag(
-        tags, tagIndex_ + indexInTags_, empty ? 0 : kTombstoneTag);
+        tags, tagIndex_ + indexInTags_, hasEmptyGroup ? 0 : kTombstoneTag);
```

Comment:
If the empty tag in the tag group is before the start index of the hash, i.e. ```(hash & sizeMask)```, is it correct to store 0? Say it hits in current tag group of index 4, and index 0-3's tag is empty, but 4-7's tag is not empty, I think it should not set 0 to index 4.
@xiaoxmeng @mbasmanova